src/dbus_api/filesystem/mod.rs

```diff
@@ -84,8 +89,6 @@ pub fn get_initial_properties(
             consts::FILESYSTEM_DEVNODE_PROP => shared::fs_devnode_prop(fs, pool_name, fs_name),
             consts::FILESYSTEM_POOL_PROP => parent,
             consts::FILESYSTEM_CREATED_PROP => shared::fs_created_prop(fs)
-        },
-        consts::PROPERTY_FETCH_INTERFACE_NAME => {},
-        consts::PROPERTY_FETCH_INTERFACE_NAME_2_1 => {}
+        }
```

Comment:
Are you sure about these removals? I know they have no properties, so it's pointless to include these, but I think we follow the D-Bus spec more closely by leaving them in. They show up in the result of GetManagedObjects(), so I think it's the right thing to do.

Reply:
A related question. Are we also supposed to send and InterfacesAdded signal when we create the root object?

Reply:
> Are you sure about these removals? I know they have no properties, so it's pointless to include these, but I think we follow the D-Bus spec more closely by leaving them in. They show up in the result of GetManagedObjects(), so I think it's the right thing to do.

I moved these to be autogenerated in the macro because they will never have any D-Bus properties associated with them. They are still sent, but I moved them so that we don't have to change it in multiple places every time we add a new FetchProperties interface version.

Reply:
> A related question. Are we also supposed to send and InterfacesAdded signal when we create the root object?

That's a very good question. I'm not entirely sure because technically it is a D-Bus object which, to me, would indicate yes. However, we will never be able to add or remove the top object during execution, only at start up and tear down, so I'm not sure how useful it is to send a signal when the D-Bus connection is just starting or shutting down. @mvollmer Are you familiar with how common/necessary this sort of thing is?

Reply:
> > Are you sure about these removals? I know they have no properties, so it's pointless to include these, but I think we follow the D-Bus spec more closely by leaving them in. They show up in the result of GetManagedObjects(), so I think it's the right thing to do.
> 
> I moved these to be autogenerated in the macro because they will never have any D-Bus properties associated with them. They are still sent, but I moved them so that we don't have to change it in multiple places every time we add a new FetchProperties interface version.

Aaah, I see it now. But it was a bit confusing, so I think this change should be moved out to a separate PR. I'll do that myself, so you can focus on your current udev experiments.

Reply:
Resolved.

Reply:
> > A related question. Are we also supposed to send and InterfacesAdded signal when we create the root object?
> 
> That's a very good question. I'm not entirely sure because technically it is a D-Bus object which, to me, would indicate yes. However, we will never be able to add or remove the top object during execution, only at start up and tear down, so I'm not sure how useful it is to send a signal when the D-Bus connection is just starting or shutting down. @mvollmer Are you familiar with how common/necessary this sort of thing is?

Is `/org/storage/stratis2` the root object?

You should not include the object that implements o.fd.Dbus.ObjectManager in the output of GetManagedObjects, nor should you send InterfacesAdded or InterfacesRemoved for it.

Reply:
Ok! We'll make sure it's not in the output of GetManagedObjects, and if it is, we'll deal w/ that.

Reply:
@mvollmer It is in the output of GetManagedObjects, because that's how dbus-rs implements it: https://github.com/stratis-storage/project/issues/216. Can you point us at a part of the D-Bus specification that indicates that this is incorrect?

Reply:
> @mvollmer It is in the output of GetManagedObjects, because that's how dbus-rs implements it: [stratis-storage/project#216](https://github.com/stratis-storage/project/issues/216). Can you point us at a part of the D-Bus specification that indicates that this is incorrect?

Oh! Here is the spec: https://dbus.freedesktop.org/doc/dbus-specification.html#standard-interfaces-objectmanager

```
All returned object paths are children of the object path implementing this interface, i.e. their object paths start with the ObjectManager's object path plus '/'. 
```

Reply:
I am not aware of any actual harm that comes from including the root in the output of GetManagedObjects, but I can imagine some endless loops if a client tries to handle nested ObjectManagers. But I don't think anybody nests ObjectManagers.

Reply:
> I am not aware of any actual harm that comes from including the root in the output of GetManagedObjects, but I can imagine some endless loops if a client tries to handle nested ObjectManagers. But I don't think anybody nests ObjectManagers.

Ok. We won't make it a high priority, but we will talk to the dbus-rs developer about it.